src/InputFilter.php

```diff
@@ -279,6 +281,11 @@ public function clean($source, $type = 'string')
 			return $result;
 		}
 
+		if (\is_object($source))
+		{
+			return (object) $this->clean((array) $source, $type);
```

Comment:
You better do it as it done for array with foreach loop.
Look a couple lines up, in theory should be enough change:
```
if (\is_array($source))
```
to
```
if (\is_array($source) || if (\is_object($source)))
```

Reply:
hmhm, no `if (\is_array($source) || if (\is_object($source)))` will be a bit different,
maybe better own loop:
```php
if (\is_object($source)))
{
   foreach ($source as $key => $value)
  {
	$result->$key = $this->clean($value, $type);
  }
 return $result;
}
```

Reply:
@Fedik Why does it need that? I have recursion, so it should be done right in my way, or not? Only difference to your way is I need one recursion level more. Or am I missing something?

Reply:
So you suggest this?

```
		if (\is_object($source))
		{
			$result = new stdClass;

			foreach ($source as $key => $value)
			{
				$result->$key = $this->clean($value, $type);
			}

			return $result;
		}
```

I think it doesn't matter if it comes before or after the `if (\is_array($source))` part, right?

Reply:
And what is better? use `new \stdClass`like in my snippet above, or use `new stdClass` and add a `use stdClass;` at the top?

Reply:
> Why does it need that? 

to avoid casting (array) and (object) back and forth :wink: